org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/indexing/SourceIndexer.java

```diff
@@ -213,6 +227,15 @@ private void purgeMethodStatements(TypeDeclaration type) {
 
 	@Override
 	public void indexResolvedDocument() {
+		// TODO We need to rebuild the DOM with binding resolution enabled
```

Comment:
What are the ramifications of having this code commented out?  Can the PR be tested without it?  You should have an issue cited here which is to fix the problem you are noting.

Reply:
This has been resolved.